### PR TITLE
feat(commands): improve /xp

### PR DESCRIPTION
**Pull request by ToastedDev**
- adds a rankcard image, which allows for a much better UI
  image
- adds `user` option, which allows the runner to specify what user to get the xp of

### CODE REVIEW

**Comment by ToastedDev** [comment]
because of this change we should probably rename `/xp` to `/rank`, but that's a debate for another day

File: bot/commands.ts [hunk 1]
```diff
@@ -129,29 +133,87 @@ const commands: Record<string, Command> = {
...
+			} else {
+			  card.setUsername("@" + interaction.user.username)
+			}
+			
+      const color = await getColor(interaction.user.displayAvatarURL({ extension: "png" }));
+      card.setStyles({
+        progressbar: {
+          thumb: {
+            style: {
+              backgroundColor: `rgb(${color[0]}, ${color[1]}, ${color[2]})`
```
Comment (GalvinPython) [inline bot/commands.ts:169]: Maybe in the future we could consider using the colour of their top role?

File: bot/commands.ts [hunk 2]
```diff
@@ -129,29 +133,87 @@ const commands: Record<string, Command> = {
...
+			
+			const card = new RankCardBuilder()
+  			.setDisplayName((interaction.member as GuildMember).displayName)
+        .setAvatar(interaction.user.displayAvatarURL()) // user avatar
+        .setCurrentXP(300) // current xp
+        .setRequiredXP(600) // required xp
+        .setLevel(2) // user level
+        .setRank(5) // user rank
+        .setOverlay(90) // overlay percentage. Overlay is a semi-transparent layer on top of the background
+        .setBackground("#23272a")
```
Comment (GalvinPython) [inline bot/commands.ts:156]: Might require a custom library but use the banner for nitro users. Or easier, use their theme for the background?

File: bot/commands.ts [hunk 3]
```diff
@@ -129,29 +133,87 @@ const commands: Record<string, Command> = {
...
+        const progressBar = createProgressBar(progress);
+        
+        await i.update({
+          embeds: [
 						quickEmbed(
 							{
 								color: 'Blurple',
 								title: 'XP',
-								description: `<@${user}> you have ${xp.xp.toLocaleString("en-US")} XP! (Level ${convertToLevels(xp.xp).toLocaleString("en-US")})`,
+								description: `<@${user}> you have ${xp.xp} XP! (Level ${convertToLevels(xp.xp)})`,
```
Comment (GalvinPython) [inline bot/commands.ts:216]: .toLocaleString() is missing. Please fix

**Review by GalvinPython — CHANGES_REQUESTED** [review]
Just need .toLocaleString() back lol

File: bot/commands.ts (hunk 1 above)
Comment (ToastedDev) [reply bot/commands.ts:169]: that would probably be better yeah, I'll implement it

File: bot/commands.ts (hunk 2 above)
Comment (ToastedDev) [reply bot/commands.ts:156]: I'm not sure if d.js has the theme color, but you are right with using banners for the background instead, we should probably consider it for the future

File: bot/commands.ts [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
 
 import client from '.';
-import { ActionRowBuilder, ButtonBuilder, ButtonStyle, type CommandInteraction, ChannelType, type APIApplicationCommandOption } from 'discord.js';
+import { ActionRowBuilder, ButtonBuilder, ButtonStyle, type CommandInteraction, ChannelType, type APIApplicationCommandOption, GuildMember, AttachmentBuilder, ComponentType } from 'discord.js';
 import { heapStats } from 'bun:jsc';
 import { getGuildLeaderboard, makeGETRequest, getRoles, removeRole, addRole, enableUpdates, disableUpdates, getCooldown, setCooldown, checkIfGuildHasUpdatesEnabled } from './utils/requestAPI';
 import convertToLevels from './utils/convertToLevels';
 import quickEmbed from './utils/quickEmbed';
+import { Font, RankCardBuilder } from 'canvacord';
+import { getColor } from 'colorthief'
```
Comment (GalvinPython) [inline bot/commands.ts]: Remove import

File: bot/commands.ts [hunk 5]
```diff
@@ -162,18 +222,20 @@ const commands: Record<string, Command> = {
 							},
 							{
 								name: 'XP Required',
-								value: `${xp.user_xp_needed_next_level.toLocaleString("en-US")} XP`,
+								value: `${xp.xp_needed_next_level} XP`,
```
Comment (GalvinPython) [inline bot/commands.ts:225]: toLocaleString() needed here as well

**Review by GalvinPython — CHANGES_REQUESTED** [review]

**Review by GalvinPython — APPROVED** [review]

**Pull request merged**